lib/transaction/transaction.js

```diff
@@ -828,17 +828,22 @@ Transaction.prototype._updateChangeOutput = function () {
   if (!_.isUndefined(this._changeIndex)) {
     this._removeOutput(this._changeIndex)
   }
+  this._changeIndex = this.outputs.length
+  this._addOutput(new Output({
+    script: this._changeScript,
+    satoshis: 0
+  }))
   var available = this._getUnspentValue()
   var fee = this.getFee()
   var changeAmount = available - fee
+  this._removeOutput(this._changeIndex)
+  this._changeIndex = undefined
   if (changeAmount > 0) {
```

Comment:
This should probably be `if (changeAmount >= 546) {` to prevent creating a change output below the dust limit.